### DISCUSSION TITLE
refresh token with loco.

### DISCUSSION THREAD

**Discussion by longjin1991** [loco-rs/loco#701] [Q&A]
how to refresh token with loco.
Can someone give me an example of refresh token with loco.
thank you.

**Comment by jondot** [comment]
the JWT token does not need a refresh in the classic OAuth way (there is no "refresh token") however, if you're worried about expiry, you can set to expire to a longer time period.
the user should re-authenticate (this is a good security practice)

if you need a token that is used for API (automation, etc), then you need to use token authentication, which sets a unique and secret token for every user record. this token does not expire unless you manually set a process to expire it, and therefore does not need refresh.
